crates/service/src/algorithms/quantization/product.rs

```diff
@@ -67,7 +67,7 @@ impl<S: G> Quan<S> for ProductQuantization<S> {
         let precomputed_table =
             serde_json::from_slice(&std::fs::read(path.join("table")).unwrap()).unwrap();
         Self {
-            dims: options.vector.dims,
+            dims: options.vector.dims.try_into().unwrap(),
```

Comment:
Let's alter `dims: u16` to `dims: u32` too.

Reply:
completed in new commit